### DISCUSSION TITLE
Is there a way to use "Users Reset Password" without using the "Email" feature

### DISCUSSION THREAD

**Discussion by MikeAlhayek** [OrchardCMS/OrchardCore#11889] [Q&A]
I am using OrchardCore in a SaaS environment. I have a default feature that I want to enable for all of my tenants that would provide `SmtpSettings` settings from the `appsettings.json` file. This way by default all of my tenants would have a way to send notification to the user without providing custom settings. Here is the only services that my feature register

```csharp
public override void ConfigureServices(IServiceCollection services)
{
    // This service is from OrchardCore.Email.Core
    services.TryAddScoped<ISmtpService, SmtpService>();

   // this is my custom-configuration provider that reads the settings from the appsettings file.
    services.TryAddTransient<IConfigureOptions<SmtpSettings>, DefaultSmtpSettingsConfiguration>();
}
```

The problem now is that I need to also enable `Users Reset Password` feature to allow users to reset their password. However, the  `Users Reset Password` feature depends on `Email` feature which is what I am trying to avoid exposing by default. The issue with the `Email` feature is that it provides a user interface to configure the EmailService settings. The Email UI is great, but in my case I don't want to enable it by default since not all tenants need to provide their own email settings. If a tenant does not want to use custom email settings, then they don't need that feature.

How can I use the ` Users Reset Password` feature and use my own implementation of `ISmtpService` instead of the build in `Email` feature? It would be nice if there is a way to separate UI from service registration.

**Comment by hishamco** [comment]
In email settings, it would be nice if the tenant doesn't set SMTP settings it fallbacks to the default tenant SMTP settings.

May be the simplest way is set the settings in each tenant - except default one - with main SMTP settings. This way the UI will populate the SMTP settings, unless you need to override them

**Reply by MikeAlhayek** [reply]
I want to have a system setting which is used by default. But if the tenant want to provide their own, then we use their custom settings.

**Comment by Skrypt** [comment]
Right now, I think you will need to use the Email feature in Orchard Core to achieve your goal. Else, we will need to think about having these settings taken from a global appsettings.json or tenant appsettings.json file from the Email feature. Right now, it is saved in the database in the SiteSettings per Tenant.

**Reply by MikeAlhayek** [reply]
Thank you for the feedback. An idea came to mind and I’ll attempt it tomorrow. The idea is to add a driver that would show 2 radio button. First is called “System” or “Default settings” and the second one would be “Custom Settings”. If the user select System, I would register read the data from the configuration file. Not sure if I’ll be able to utilize existing driver or not. I’ll update the post after I try it. 

But generally speaking, I think it would be great to be able to split the UI features into a separate feature in OC. Another example, I want to use OrchardCore.Users module but I don’t want the UI. I want to have my own UI. Currently there is no easy way to turn off the UI only.
